assets/src/edit-story/panels/size.js

```diff
@@ -6,74 +6,84 @@ import PropTypes from 'prop-types';
-		evt.preventDefault();
-	};
+
+	const handleSubmit = useCallback(
+		( event ) => {
+			onSetProperties( state );
+			event.preventDefault();
+		},
+		[ state, onSetProperties ],
+	);
+
+	const handleRatio = useCallback( () => {
+		setLockRatio( ! lockRatio );
+	}, [ lockRatio, setLockRatio ] );
+
+	const handleWidth = useCallback( ( value ) => {
+		const ratio = width / height;
+		const newWidth = isNaN( value ) || value === '' ? '' : parseFloat( value );
+		setState( {
+			...state,
+			width: newWidth,
+			height: typeof newWidth === 'number' && lockRatio ? newWidth / ratio : height,
+		} );
+	}, [ width, height, state, setState, lockRatio ] );
+
+	const handleHeight = useCallback( ( value ) => {
+		const ratio = width / height;
+		const newHeight = isNaN( value ) || value === '' ? '' : parseFloat( value );
+		setState( {
+			...state,
+			height: newHeight,
+			width: typeof newHeight === 'number' && lockRatio ? newHeight * ratio : width,
+		} );
+	}, [ width, height, state, setState, lockRatio ] );
+
 	return (
 		<Panel onSubmit={ handleSubmit }>
-			<Title>
-				{ __( 'Size', 'amp' ) }
-			</Title>
-			<InputGroup
-				label={ __( 'Width', 'amp' ) }
-				value={ state.width }
-				isMultiple={ width === '' }
-				onChange={ ( value ) => {
-					const ratio = width / height;
-					const newWidth = isNaN( value ) || value === '' ? '' : parseFloat( value );
-					setState( {
-						...state,
-						width: newWidth,
-						height: typeof newWidth === 'number' && lockRatio ? newWidth / ratio : height,
-					} );
-				} }
-				postfix={ _x( 'px', 'pixels, the measurement of size', 'amp' ) }
-				disabled={ isFullbleed }
-			/>
-			<InputGroup
-				label={ __( 'Height', 'amp' ) }
-				value={ state.height }
-				isMultiple={ height === '' }
-				onChange={ ( value ) => {
-					const ratio = width / height;
-					const newHeight = isNaN( value ) || value === '' ? '' : parseFloat( value );
-					setState( {
-						...state,
-						height: newHeight,
-						width: typeof newHeight === 'number' && lockRatio ? newHeight * ratio : width,
-					} );
-				} }
-				postfix={ _x( 'px', 'pixels, the measurement of size', 'amp' ) }
-				disabled={ isFullbleed }
-			/>
-			<InputGroup
-				type="checkbox"
-				label={ __( 'Keep ratio', 'amp' ) }
-				value={ lockRatio }
-				isMultiple={ false }
-				onChange={ ( value ) => {
-					setLockRatio( value );
-				} }
-				disabled={ isFullbleed }
-			/>
+			<Row>
+				<PrefixInput
+					label={ __( 'W', 'amp' ) }
+					value={ state.width }
+					isMultiple={ width === '' }
+					onChange={ handleWidth }
+					disabled={ isFullbleed }
+				/>
+				<SpaceSpan />
+				<LockSwitch
+					label={ __( 'Lock Icon', 'amp' ) }
+					checked={ lockRatio }
+					onClick={ handleRatio }
+				/>
+				<SpaceSpan />
+				<PrefixInput
+					label={ __( 'H', 'amp' ) }
+					value={ state.height }
+					isMultiple={ height === '' }
+					onChange={ handleHeight }
+					disabled={ isFullbleed }
+				/>
+			</Row>
```

Comment:
Much nicer output! This is what we should always strive for - reads exceptionally clearly.